src/main/java/teammates/ui/webapi/action/PutCourseStudentDetailsEditAction.java

```diff
@@ -66,8 +66,8 @@ public ActionResult execute() throws EntityNotFoundException {
         student.comments = SanitizationHelper.sanitizeTextField(student.comments);
 
         try {
+            student.nullToEmptyString();
             StudentAttributes originalStudentAttribute = logic.getStudentForEmail(courseId, studentEmail);
-            student.updateWithExistingRecord(originalStudentAttribute);
```

Comment:
> @xpdavid is this the right thing to do?

I don't think so. Actually, this is the only place where `updateWithExistingRecord` is used. It is imprudent to change to `student.nullToEmptyString();` because L72 & L73 do validation based on the change of team and section.

Reply:
Reason why I changed them to empty string is because further down the file there is this logic to update students:
```
                logic.updateStudentCascade(
                        StudentAttributes.updateOptionsBuilder(courseId, studentEmail)
                                .withName(student.name)
                                .withNewEmail(student.email)
                                .withTeamName(student.team)
                                .withSectionName(student.section)
                                .withComment(student.comments)
                                .build());
```
For example, if the `name` field is null, then this section of the code (refer below) will return the error message `Field to update is null` to the user.
```
            public Builder withName(String name) {
                Assumption.assertNotNull(Const.StatusCodes.UPDATE_OPTIONS_NULL_INPUT, name);

                updateOptions.nameOption = UpdateOption.of(name);
                return this;
            }
```
However, I want `InvalidParametersException` to be thrown instead, so a more accurate and informed error message would be returned to the user.

Reply:
I understand the concern. However, you need to fix the side-effect (i.e. L72 & 73) of changing this.

Reply:
Is excluding the null check for both section and team in the `nullToEmptyString` function good enough? 

Since `isSectionChanged` and `isTeamChanged` checks if they are null already.